src/dune_rules/merlin/ocaml_index.ml

```diff
@@ -63,7 +81,8 @@ let cctx_rules cctx =
       ; A "-o"
       ; Target fn
       ; Deps modules_deps
-      ; Dyn (Resolve.Memo.read additional_libs)
```

Comment:
Could you go back to using `Dyn`?

Reply:
`Resolve.args`  felt more idiomatic, but I can switch back of course. Is there a non-cosmetic reason which make using ` Dyn`  better ? Because it pushes the memos a bit deeper ?

Reply:
There's indeed a non cosmetic reason. In the previous code, we'd compute the dependencies when we needed to run the action. While in your code, we compute the dependencies when loading the rules. A simplified summary of how dune executes a rule will be helpful here. It goes like this:

1. Given the target we're building, find the directory of this target
2. In this directory, load the rules.
3. Find the rule that corresponds to the target
4. Compute the dependencies of the rule
5. Build the dependencies of the rule
6. Execute the rule to build the target.

Your change pushes work from step 4 to step 2. This is unnecessary work when we're loading the rules in the directory to build some other rule.

Reply:
Thanks a lot for the detailed explanation, I made the change in [cf0a087](https://github.com/ocaml/dune/pull/10623/commits/cf0a0873a636d829e12b7b8f122a4ff10d88ac46) and I will keep it in mind for my future contributions.